sound/soc/intel/boards/sof_es8336.c

```diff
@@ -326,8 +398,13 @@ static int sof_es8336_probe(struct platform_device *pdev)
 	}
 	log_quirks(dev);
 
+	if (soc_intel_is_apl() || soc_intel_is_glk())
+		dmic_be_num = 0;
```

Comment:
I think this would be better with a quirk. Nothing prevents the use of DMIC on APL/GLK, this is not a property of the SoC but a board integration decision.

Reply:
Hi Pierre, 
Thanks for your comments. I prefer to add a quirk for GLK device which doesn't use pch-dmic, but unfortunately my GLK device is from ODM, it lacks of detailed vendor or board info from DMI. Moreover, this GLK board design may be referred by other OEM too. So i added a quirk (SOF_ES8336_ENABLE_DMIC) for the device which uses pch-dmic. Please help to review the new update, thanks.